### PR TITLE
Create upload release workflow

### PR DESCRIPTION
**Pull request by lisguo**
*Description of changes:*
Runs the integration tests and on success, pushes the staging artifact to release.

*Tests*
Ran workflow on push to branch and saw a successful run. Used a test tag
https://github.com/aws/amazon-cloudwatch-agent-operator/actions/runs/6433487066/job/17470648145

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/integration-test.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@ on:
       - '.github/**'
       - '!.github/workflows/integration-test.yml'
   workflow_dispatch:
+  workflow_call:
```
Comment (jefchien) [inline .github/workflows/integration-test.yml:16]: What are we using workflow_call for?

File: .github/workflows/upload-release.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+name: Upload Release Artifact
+env:
+  # Use terraform assume role for uploading to ecr
+  AWS_ASSUME_ROLE: ${{ secrets.TERRAFORM_AWS_ASSUME_ROLE }}
+  ECR_OPERATOR_STAGING_IMAGE: ${{ secrets.ECR_OPERATOR_STAGING_IMAGE }}
+  ECR_OPERATOR_RELEASE_IMAGE: ${{ secrets.ECR_OPERATOR_RELEASE_IMAGE }}
+
+on:
+  workflow_dispatch:
```
Comment (jefchien) [inline .github/workflows/upload-release.yml:12]: Are we going to tag the releases? Could this be triggered on release?

File: .github/workflows/upload-release.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (jefchien) [inline .github/workflows/upload-release.yml:1]: Do we need to add the copyright header in the other workflows as well?

**Review by jefchien — APPROVED** [review]

File: .github/workflows/integration-test.yml (hunk 1 above)
Comment (lisguo) [reply .github/workflows/integration-test.yml:16]: this allows the upload-release workflow to call the integ test as a pre-step to upload

File: .github/workflows/upload-release.yml (hunk 2 above)
Comment (lisguo) [reply .github/workflows/upload-release.yml:12]: I have an action to come up with a versioning strategy, but for now it is just tagged as latest which is part of the secret

File: .github/workflows/upload-release.yml (hunk 3 above)
Comment (lisguo) [reply .github/workflows/upload-release.yml:1]: yeah good point, actually this header should be referencing apache

File: .github/workflows/integration-test.yml (hunk 1 above)
Comment (jefchien) [reply .github/workflows/integration-test.yml:19]: Oh right. Reusable workflows.

**Pull request merged**